utils/crash_triage/triage_crashes.sh

```diff
@@ -90,8 +90,9 @@ for crash in $DIR/crashes/id:*; do
 
   for a in $@; do
 
-    if [ "$a" = "@@" ] ; then
-      use_args="$use_args $crash"
+    if echo "$a" | grep -qF '@@'; then
```

Comment:
I think this can go wrong if $a is e.g. -e... as this then interpreted as an option to echo?

Reply:
$a is quoted here to prevent that?
I took the line almost verbatim from afl-cmin.bash https://github.com/AFLplusplus/AFLplusplus/blob/e5bdba4b9f22c0f2e4ff60ffc9bfa8fbaeb586d2/afl-cmin.bash#L173

Reply:
On second thought you're right - both of these are probably wrong.
If we were allowed to assume bash, a safer solution would be to do `grep -qF '@@' <<< "$a"` but I'm not really sure how to do this in plain old `sh`.

Reply:
Upon further testing, I think it is safe due to the quote. It at least takes the correct branch in the if statement:

```sh
$ if echo "-e@@" | grep -qF "@@"; then echo "found @@"; else echo "Nope"; fi
found @@
$ if echo "-e @@" | grep -qF "@@"; then echo "found @@"; else echo "Nope"; fi
found @@
$ if echo "-e" | grep -qF "@@"; then echo "found @@"; else echo "Nope"; fi
Nope
$ if echo "-n" | grep -qF "@@"; then echo "found @@"; else echo "Nope"; fi
Nope
$ if echo "--version" | grep -qF "@@"; then echo "found @@"; else echo "Nope"; fi
Nope
```

Reply:
POSIX `/bin/echo` and the more portable `/bin/printf` do not support any options. But GNU `/bin/echo` and the built-in command `echo` in bash both support `-n`, `-e`, `-E` making it non trivial to output a string with `-[neE]+`, e.g. `-nennen`.
So quoting `-e` does not prevent `echo` from consuming it instead of echoing it, which would spoil the targets argument.
Also echo adds an unwanted newline.
How about this instead?
```
  for a in $@; do

    case "$a" in
      *@@*)
      unset use_stdio
      use_args="$use_args `printf %s "$a" | sed -e 's<@@<'$crash'<g'`"
      ;;
      *)
      use_args="$use_args $a"
      ;;
    esac

  done

```

Reply:
I like the use of `case` to do the pattern matching!
I'm also a fan of assuming files don't have `<` in their names, although technically linux allows anything except slash and null.
Since you mentioned printf, maybe we can use `printf %q "$crash"` to escape the filename and pass that in rather than the crazy sed slash escape that I made?

Regardless, I have shown in my earlier comment that the original code in the PR actually does what is expected. It accurately determines that "-e" and friends are _not @@_ and sends them to the other path. This code also matches what is done in cmin.bash:
https://github.com/AFLplusplus/AFLplusplus/blob/e5bdba4b9f22c0f2e4ff60ffc9bfa8fbaeb586d2/afl-cmin.bash#L173

My recommendation is to take this PR as-is and then (if desired) make another one to address potential clean-ups with both triage_crashes.sh and cmin.bash.